src/transports/janus_websockets.c

```diff
@@ -1347,6 +1347,9 @@ static int janus_websockets_common_callback(
 					incoming_curr += error.position;
 					JANUS_LOG(LOG_HUGE, "[%s-%p] Parsed JSON message - consumed %zu/%zu bytes\n",
 						log_prefix, wsi, (size_t)(incoming_curr - ws_client->incoming), incoming_length);
+					/* Trailing whitespace after the last message results in invalid JSON error */
+					while isspace(*incoming_curr)
+						incoming_curr++;
```

Comment:
This sounds like a dangerous check to me, though, since it's increasing the pointer and dereferencing it without ever checking if it got to the end and beyond (i.e., after `incoming_end`). This will crash if `incoming_curr` leaves the boundaries of the allocated space.

Reply:
I considered this for a moment, but couldn't see how this could overflow. Still, I agree that it's better to be safe than sorry, so I pushed an update. Thanks for the feedback!

Reply:
Thanks! I'll make a couple of quick ckecks later (or tomorrow morning) and in case I can't spot any issue, I'll merge both PRs (thanks for providing a backported version too) :+1: